### PR TITLE
Documentation added for digital signature verification

### PR DESCRIPTION
**Pull request by abhupadh**
Adding documentation describing the I/O Events new digital signature feature

## Description
This ticket
-  adds documentation for digital signature. 
-  documentation describing the complete digital signature signing process of I/O Events
- describing the additional verification of the target recipient
- documentation describing the customer experience with the changed event signing process

See ticket CI-4936 and CI-4937 for details. 
Also follow [this](https://wiki.corp.adobe.com/display/AdobeCloudPlatform/Webhook+Digital+Signatures+-+Implementation+Plan) wiki here for additional information

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/guides/index.md [hunk 1]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
...
+
+Once you have the public keys fetched as plain text you can now verify the digital signatures by following the steps as below
+
+- decrypt the message digest using the public key
+- compute the hash message digest of the event payload (available in the webhook request body) using the same hash function algorithm `rsa-sha256` used by I/O Events during signing
+- now validate each signature by comparing 
+  - the message digest computed by hashing 
+  - and the digest received after decrypting the signature using the public key
+
+Refer to [this](https://github.com/adobe/aio-lib-events/blob/4694ef1b4f19c63131d506e1abfe7a3e555e2914/src/index.js#L491) signature verification method of the events sdk to understand how you can implement the same validation for your webhook app.
```
Comment (francoisledroff) [inline src/pages/guides/index.md:365]: I'm surprised to see why still use the clientSecret

**Review by francoisledroff — CHANGES_REQUESTED** [review]
a few comments/questions

File: src/pages/guides/index.md [hunk 2]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
...
+            "lmesMetadata:assetlanguage": ""
+        },
+        "xdmEventEnvelope:objectType": "xdmAsset"
+    }
+}
+```
+Sample XDM format asset event payload after
+```json
+{
+    "recipient_client_id": "webhook_registration_client_id",
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: done

File: src/pages/guides/index.md [hunk 3]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
...
+- the digest is then encrypted using the I/O Events private key to generate the digital signature
+
+I/O Events will send the 2 digital signatures as webhook request headers and they will be available via the header fields
+`x-adobe-digital-signature-1` and `x-adobe-digital-signature-2` respectively. 
+
+I/O Events will also send 2 public keys corresponding to the private keys used to generate the digital signatures. These public keys which are stored in Cloud Front are publicly accessible using the Cloud Front urls which look like `https://xxxxx.cloudfront.net/pub-key-1.pem`. I/O Events will send the public keys urls for both the keys and they will be available via the the request header fields `x-adobe-public-key1-url` and  `x-adobe-public-key2-url` respectively.
+
+As mentioned earlier, I/O Events transforms your event payload and adds an additional json field `recipient_client_id` to your payload. See the sample payload before and after transformation that I/O Events will send to your webhook.
+
+Sample XDM format asset event payload before
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: removed

File: src/pages/guides/index.md [hunk 4]
```diff
@@ -244,7 +244,7 @@ This signature or "message authentication code" is computed using a cryptographi
 Upon receiving a request, you should repeat this calculation and compare the result to the value in the `x-adobe-signature` header, and reject the request unless they match. Since the client secret is known only by you and Adobe I/O Events, this is a reliable way to verify the authenticity of the request.
 
 <InlineAlert variant="info" slots="text"/>
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: done

File: src/pages/guides/index.md (hunk 1 above)
Comment (abhupadh) [reply src/pages/guides/index.md]: > I'm surprised to see why still use the clientSecret

This link will be updated to point to the new signature verification method of the sdk which is being done in [this](https://github.com/adobe/aio-lib-events/pull/17) PR. This PR will be merged only after making this link update

Comment (abhupadh) [reply src/pages/guides/index.md]: > I would move this section right after the steps to verify (L354)

I don't think so, as all the steps above this line accounts for the complete verification steps. So, I think having this sentence after that would link the context better.

File: src/pages/guides/index.md [hunk 5]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
 if (request.header('x-adobe-signature') !== hmac.digest('base64')) {
   throw new Error('x-adobe-signature HMAC check failed')
 }
-```
\ No newline at end of file
+```
+
+**Improved Security Verification for Webhook Events**
+
+For a more robust and reliable verification, Adobe I/O Events will add below security validations for events delivered to your webhook.
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: done. used present

File: src/pages/guides/index.md [hunk 6]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
 if (request.header('x-adobe-signature') !== hmac.digest('base64')) {
   throw new Error('x-adobe-signature HMAC check failed')
 }
-```
\ No newline at end of file
+```
+
+**Improved Security Verification for Webhook Events**
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: done

File: src/pages/guides/index.md [hunk 7]
```diff
@@ -256,4 +256,113 @@ hmac.update(raw_request_body)
...
+- decrypt the message digest using the public key
+- compute the hash message digest of the event payload (available in the webhook request body) using the same hash function algorithm `rsa-sha256` used by I/O Events during signing
+- now validate each signature by comparing 
+  - the message digest computed by hashing 
+  - and the digest received after decrypting the signature using the public key
+
+Refer to [this](https://github.com/adobe/aio-lib-events/blob/4694ef1b4f19c63131d506e1abfe7a3e555e2914/src/index.js#L491) signature verification method of the events sdk to understand how you can implement the same validation for your webhook app. 
+
+<InlineAlert variant="info" slots="text"/>
+Adobe strongly encourages validating your webhook deliveries using this mechanism now to avoid processing "events" received from malicious third-party actors and make sure your webhook continues to receive events.
```
[the remark this answers is not in the document]
Comment (abhupadh) [reply src/pages/guides/index.md]: done

**Comment by abhupadh** [comment]
@ledroff/ @shikhartanwar : let me know if the changes look good to be merged now. thanks

File: src/pages/guides/index.md [hunk 8]
```diff
@@ -235,17 +235,64 @@ For Creative Cloud Asset events, you'll need to add the Creative Cloud Libraries
...
+
+Once you have the public keys fetched as plain text you can now verify the digital signatures by following the steps as below
+
+- decrypt the message digest using the public key
+- compute the hash message digest of the event payload (available in the webhook request body) using the same hash function algorithm `rsa-sha256` used by I/O Events during signing
+- now validate each signature by comparing 
+  - the message digest computed by hashing 
+  - and the digest received after decrypting the signature using the public key
+
+Refer to [this](https://github.com/adobe/aio-lib-events/blob/4694ef1b4f19c63131d506e1abfe7a3e555e2914/src/index.js#L491) signature verification method of the events sdk to understand the above signature validation steps for your webhook app.
```
Comment (shikhartanwar) [inline src/pages/guides/index.md:283]: We would still need to change this link right?

Comment (abhupadh) [reply src/pages/guides/index.md:283]: yes, once the sdk changes gets merged, will use the link and then merge

Comment (francoisledroff) [reply src/pages/guides/index.md:283]: ok then let us know when the link is updated and we'll review-again/approve

Comment (abhupadh) [reply src/pages/guides/index.md]: updated.

File: src/pages/guides/index.md [hunk 9]
```diff
@@ -280,7 +280,7 @@ Once you have the public keys fetched as plain text you can now verify the digit
   - the message digest computed by hashing 
   - and the digest received after decrypting the signature using the public key
 
-Refer to [this](https://github.com/adobe/aio-lib-events/blob/4694ef1b4f19c63131d506e1abfe7a3e555e2914/src/index.js#L491) signature verification method of the events sdk to understand the above signature validation steps for your webhook app.
+Refer to [this](https://github.com/adobe/aio-lib-events/blob/master/src/index.js#L516) signature verification method of the events sdk to understand the above signature validation steps for your webhook app.
```
Comment (shikhartanwar) [inline src/pages/guides/index.md:283]: Links pointing to master get outdated very easily. Can we release the SDK and instead have the link to a tagged version of it?
Like https://github.com/adobe/aio-lib-events/blob/1.1.0/src/index.js

**Review by shikhartanwar — APPROVED** [review]

**Pull request merged**